ci-generate-build-n-test.js

```diff
@@ -0,0 +1,127 @@
+const fs = require("fs");
+
+const services = [
+  "auditlogs",
+  "create-artifact-worker",
+  "deployments",
+  "deviceauth",
+  "deviceconfig",
+  "deviceconnect",
+  "devicemonitor",
+  "inventory",
+  "mender-gateway",
+  "tenantadm",
+  "useradm",
+  "workflows",
+];
+
+const pipelineTemplate = (jobs) => `stages:
+  - build
+  - test
+${jobs.join("")}
+`;
+
+const getChangesRule = ({
+  RULES_CHANGES_COMPARE_TO_REF,
+  service,
+}) => `- changes:
+        paths: ["backend/services/${service}/**/*", "backend/pkg/**/*"]
+        compare_to: "${RULES_CHANGES_COMPARE_TO_REF}"`;
+
+const buildTemplate = ({
+  CI_COMMIT_REF_NAME,
+  CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX,
+  CI_PIPELINE_ID,
+  CI_REGISTRY_IMAGE,
+  DOCKER_PUBLISH_COMMIT_TAG,
+  RULES_CHANGES_COMPARE_TO_REF,
+  service,
+}) => `
+build:${service}:docker-multiplatform:
```

Comment:
Building services in different jobs, although parallel is more wasteful as it cannot leverage the local go cache.